lib/accesslists/validate.go

```diff
@@ -315,3 +315,73 @@ func exceedsMaxDepth(
 		return totalDepth > accesslist.MaxAllowedDepth, nil
 	}
 }
+
+func maxDepthDownwards(
```

Comment:
I think this warrants a comment explaining what do we want to check and a test.

Reply:
It isi used and this is just PR just moved this funtion from one place to another.